producer/src/contract/contract_test.go

```diff
@@ -58,24 +75,60 @@ func TestMakeYield(t *testing.T) {
 
 /* Simple test to make sure insertion is working */
 func TestInsertYield(t *testing.T) {
-	setUpDB("testDB.db")
+	err := setUpDB("testDB.db")
+	if err != nil {
+		t.Errorf("Failed to set up database")
+		log.Fatal(err)
+	}
 	defer tearDown("testDB.db")
 	testPubKey := generatePubKey()
-	testYield := MakeYield(testPubKey, 10000000)
+	testYield := MakeYield(&testPubKey, 10000000)
 	contractHash := block.HashSHA256([]byte{'b', 'l', 'k', 'c', 'h', 'a', 'i', 'n'})
-	err := InsertYield(testYield, "testDB.dat", 35, contractHash, 1)
-	if err != nil {
+	err2 := InsertYield(testYield, "testDB.db", 35, contractHash, 1)
+	if err2 != nil {
 		t.Errorf("Failed to insert yield")
+		log.Fatal(err2)
 	}
+
+	dbConn, _ := sql.Open("sqlite3", "testDB.db")
+	//check if a row is actually inserted
+	sqlQuery := `SELECT * FROM uy;`
+	rows, err2 := dbConn.Query(sqlQuery)
+	if err2 != nil {
+		t.Errorf("failed to query db")
+	}
+
+	var dbHeight uint32
+	var dbContract string
+	var dbIndex uint8
+	var dbHolder string
+	var dbValue uint64
+
+	if rows.Next() {
+		if err3 := rows.Scan(&dbHeight, &dbContract, &dbIndex, &dbHolder, &dbValue); err3 != nil {
+			log.Fatal(err3)
+		}
+		//check if each value is correct
+		assert.Equal(t, uint32(35), dbHeight, "The height value in the database is wrong")
+		assert.Equal(t, hex.EncodeToString(contractHash), dbContract, "The contract hash in database is wrong")
+		assert.Equal(t, uint8(1), dbIndex, "The yield index in database is wrong")
+		assert.Equal(t, hex.EncodeToString(testYield.Recipient), dbHolder, "The holder in database is wrong")
+		assert.Equal(t, testYield.Value, dbValue, "The value in the database is wrong")
+	} else {
+		t.Errorf("InsertYield failed to insert row into database")
+	}
+
+	rows.Close()
+	dbConn.Close()
 }
 
 /* Tests both serialization and deserialization */
 func TestYieldSerialization(t *testing.T) {
 	testPubKey := generatePubKey()
-	expected := MakeYield(testPubKey, 200000)
+	expected := MakeYield(&testPubKey, 200000)
 	serialized := expected.Serialize()
 	deserialized := DeserializeYield(serialized)
-	if !cmp.Equal(deserialized, expected) {
+	if !reflect.DeepEqual(deserialized, expected) {
```

Comment:
Why was this changed? cmp.Equal is better for struct comparison than reflect

Reply:
For some reason reflect.DeepEqual works, and I also checked the values of the struct and they were equal, but cmp.Equal still returns the opposite.

Reply:
Also according to the document cmp.Equal is similar to reflect.DeepEqual.

Reply:
If they are similar then they should both pass

Reply:
Yeah I think I made a mistake (probably missing a !) initially, because both ways work